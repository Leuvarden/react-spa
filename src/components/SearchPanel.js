import React, { Component } from 'react';
import Logo from './Logo';

export default class SearchPanel extends Component {
    render () {
        const inputElementWithLabel = () => {
            return (
                <div className="search-panel__form">
                    <label htmlFor="searchPanelInput">Find your movie</label>
                    <input type="text" id="searchPanelInput"/>
                </div>
            );          
        };

        const filterButton = (name) => {
            return (
                <button className="search-panel__button-filter">{name}</button>
            );
        };

        return ( 
            <section className="search-panel">
                <div className="search-panel__container">
                    {Logo}
                    {inputElementWithLabel()}
                    <div className="search-panel__action-elements">
                        <div className="search-panel__filter">
                            <label id="filterDescription">Search by</label>
                            {filterButton('Title')}
                            {filterButton('Genre')}
                        </div>
                        <button className="search-panel__button-search">Search</button>
                    </div>
                </div>
            </section>
        );
    }
}