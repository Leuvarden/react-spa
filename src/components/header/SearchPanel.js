import React, { Component } from 'react';
import Logo from './../main/Logo';
import SearchForm from './SearchForm';
import SearchFilter from './SearchFilter';
import './../../styles/search-panel.scss';

class SearchPanel extends Component {
    render () {
        return (
            <section className="search-panel">
                <div className="search-panel__container">
                    <Logo />
                    <SearchForm />
                    <SearchFilter />
                </div>
            </section>
        );
    }
}

export default SearchPanel;