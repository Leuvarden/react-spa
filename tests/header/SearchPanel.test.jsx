import React from 'react';
import SearchPanel from './../../src/components/header/SearchPanel';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

/* eslint-disable no-undef */
describe ('SearchPanel', () => {

    const exampleStore = () => ({data: []});
    const store = createStore(exampleStore);

    it('renders as expected', () => {
        const panel = render(
            <Provider store={store}>
                <SearchPanel />
            </Provider>
        );
    
        expect(panel).toMatchSnapshot();
    });

});