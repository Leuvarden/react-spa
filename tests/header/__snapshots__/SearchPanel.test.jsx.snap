// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel renders as expected 1`] = `
<section
  class="search-panel"
>
  <div
    class="search-panel__container"
  >
    <div
      class="logo"
    >
      netflixroulette
    </div>
    <div
      class="search-panel__form"
    >
      <label
        for="searchPanelInput"
      >
        Find your movie
      </label>
      <input
        id="searchPanelInput"
        type="text"
      />
    </div>
    <div
      class="search-panel__filter"
    >
      <label
        id="filterDescription"
      >
        Search by:
        <button
          class="search-panel__button-filter"
        >
          Title
        </button>
        <button
          class="search-panel__button-filter"
        >
          Genre
        </button>
      </label>
      <button
        class="search-panel__button-search"
      >
        Search
      </button>
    </div>
  </div>
</section>
`;
